metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java

```diff
@@ -153,4 +154,51 @@ void checkKPIDisabledByDefault() {
                         .findAny();
         assertThat("In-flight concurrent gauge metric exists", inflightRequests.isPresent(), is(isKPIEnabled));
     }
+
+    @Test
+    void checkMetricsForExecutorService() {
+
+        String jsonKeyForCompleteTaskCountInThreadPool =
+                "executor-service.completed-task-count;poolIndex=0;supplierCategory=helidon-thread-pool-1;supplierIndex=0";
+
+        WebClientRequestBuilder metricsRequestBuilder = webClientBuilder
+                .build()
+                .get()
+                .accept(MediaType.APPLICATION_JSON)
+                .path("metrics/vendor");
+
+        WebClientResponse response = metricsRequestBuilder
+                .submit()
+                .await();
+
+        assertThat("Normal metrics/vendor URL HTTP response", response.status().code(), is(200));
+
+        JsonObject metrics = response.content().as(JsonObject.class).await();
+
+        int completedTaskCount =
+                metrics.getInt(jsonKeyForCompleteTaskCountInThreadPool);
+        assertThat("Completed task count before accessing slow endpoint", completedTaskCount, is(0));
+
+        WebClientResponse slowGreetResponse = webClientBuilder
+                .build()
+                .get()
+                .accept(MediaType.TEXT_PLAIN)
+                .path("greet/slow")
+                .submit()
+                .await();
+
+        assertThat("Slow greet access response status", slowGreetResponse.status().code(), is(200));
+
+        WebClientResponse secondMetricsResponse = metricsRequestBuilder
+                .submit()
+                .await();
```

Comment:
There is some internal timeout in the webclient but it is always better to use `await` with some duration

Reply:
Fixed.